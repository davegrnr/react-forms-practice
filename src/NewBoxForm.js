import React, {useState} from 'react';

const NewBoxForm = ({ addBox }) => {
    const INITIAL_STATE = {
        color: '',
        height: '',
        width: '',
    }

    const [formData, setFormData] = useState(INITIAL_STATE)
    const handleChange = (e) => {
        const {name, value} = e.target
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        addBox({...formData})
        setFormData(INITIAL_STATE)
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="color">Color </label>
            <input 
                id="color"
                type="text"
                name="color"
                placeholder="Box Color"
                value={formData.color}
                onChange={handleChange}
            />
            <label htmlFor="height"> Height </label>
            <input 
                id="height"
                type="text"
                name="height"
                placeholder="Box Height"
                value={formData.height}
                onChange={handleChange}
            /> 
            <label htmlFor="width"> Width </label>
            <input 
                id="width"
                type="text"
                name="width"
                placeholder="Box Width"
                value={formData.width}
                onChange={handleChange}
            />
                <br></br>
            <button style={{margin: '10px'}}>Box Time!</button>
        </form>
    )
}

export default NewBoxForm