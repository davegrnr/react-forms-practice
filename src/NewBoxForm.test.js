import React from "react";
import { fireEvent, render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

it("renders without crashing", () => {
    render(<NewBoxForm />)
})

it("matches a snapshot", () => {
    const { asFragment } = render(<NewBoxForm />)
    expect(asFragment()).toMatchSnapshot()
})



// it("", () => {
    
// })