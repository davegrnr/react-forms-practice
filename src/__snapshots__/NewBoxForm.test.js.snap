// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="color"
    >
      Color 
    </label>
    <input
      id="color"
      name="color"
      placeholder="Box Color"
      type="text"
      value=""
    />
    <label
      for="height"
    >
       Height 
    </label>
    <input
      id="height"
      name="height"
      placeholder="Box Height"
      type="text"
      value=""
    />
    <label
      for="width"
    >
       Width 
    </label>
    <input
      id="width"
      name="width"
      placeholder="Box Width"
      type="text"
      value=""
    />
    <br />
    <button
      style="margin: 10px;"
    >
      Box Time!
    </button>
  </form>
</DocumentFragment>
`;
